lib/passwd.ml

```diff
@@ -26,24 +26,36 @@ type passwd_t
 
 let passwd_t : passwd_t structure typ = structure "passwd"
 
-let pw_name   = field passwd_t "pw_name" string
-let pw_passwd = field passwd_t "pw_passwd" string
+let pw_name   = field passwd_t "pw_name" (ptr char)
+let pw_passwd = field passwd_t "pw_passwd" (ptr char)
 let pw_uid    = field passwd_t "pw_uid" uint32_t
 let pw_gid    = field passwd_t "pw_gid" uint32_t
-let pw_gecos  = field passwd_t "pw_gecos" string
-let pw_dir    = field passwd_t "pw_dir" string
-let pw_shell  = field passwd_t "pw_shell" string
+let pw_gecos  = field passwd_t "pw_gecos" (ptr char)
+let pw_dir    = field passwd_t "pw_dir" (ptr char)
+let pw_shell  = field passwd_t "pw_shell" (ptr char)
 
 let () = seal passwd_t
 
+let ptr_char_to_string p =
```

Comment:
Do we have to copy the C string because we don't know it's lifetime and it will be de-allocted from C at some point? But we do know that it won't be de-allocated before we have made the copy? Maybe this would benefit from a comment about these assumptions.

Reply:
I will add a comment. I was relying on the commit message to try and be as explicit and exhaustive as possible